assets/src/edit-story/components/panels/stylePreset/header.js

```diff
@@ -0,0 +1,201 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import styled, { css } from 'styled-components';
+import { rgba } from 'polished';
+import { useContext } from 'react';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import PropTypes from 'prop-types';
+import { ReactComponent as Edit } from '../../../icons/edit_pencil.svg';
+import { ReactComponent as Add } from '../../../icons/add_page.svg';
+import { ReactComponent as Arrow } from '../../../icons/arrow.svg';
+import panelContext from '../panel/context';
+
+const Header = styled.h2`
+  border: 0 solid ${({ theme }) => theme.colors.bg.v9};
+  color: ${({ theme }) => theme.colors.fg.v1};
+  margin: 0;
+  position: relative;
+  display: flex;
+  flex-direction: column;
+  justify-content: flex-start;
+  align-items: stretch;
+  user-select: none;
+`;
+
+const HeaderButton = styled.div.attrs({ role: 'button' })`
+  color: inherit;
+  border: 0;
+  padding: 10px 20px;
+  background: transparent;
+  display: flex;
+  justify-content: space-between;
+  align-items: center;
+  cursor: pointer;
+`;
+
+const Heading = styled.span`
+  color: inherit;
+  margin: 0;
+  font-weight: 500;
+  font-size: 14px;
+  line-height: 19px;
+`;
+
+const HeaderActions = styled.div`
+  display: flex;
+  align-items: center;
+`;
+
+const buttonCSS = css`
+  background: transparent;
+  width: 30px;
+  height: 28px;
+  border: none;
+  color: ${({ theme }) => rgba(theme.colors.fg.v1, 0.84)};
+  cursor: pointer;
+  padding: 0;
+`;
+
+const AddColorPresetButton = styled.button`
+  ${buttonCSS}
+  svg {
+    width: 26px;
+    height: 28px;
+  }
+`;
+
+const Collapse = styled.button`
+  ${buttonCSS}
+  svg {
+    width: 28px;
+    height: 28px;
+    ${({ isCollapsed }) => isCollapsed && `transform: rotate(.5turn);`}
+  }
+`;
+
+const ExitEditMode = styled.button`
+  ${buttonCSS}
+  color: ${({ theme }) => theme.colors.fg.v1};
+  font-size: 12px;
+  line-height: 14px;
+  padding: 7px;
+  height: initial;
+`;
+
+const EditModeButton = styled.button`
+  ${buttonCSS}
+  height: 20px;
+  svg {
+    width: 16px;
+    height: 20px;
+  }
+`;
+
+function PresetsHeader({
```

Comment:
This is duplicating the panel title behavior without reusing it? That seems like poor architecture?

Reply:
Yep, was reusing the panel title initially, however, separated it since was trying out some accessibility options with attribute/tag changes.

Will try to reuse it again, however, there are some things to confirm first, for example:
In case of all the other panels, the full title gets outlined when tabbing. However, this panel has more actions than just collapsing and not collapsing -- should in this case the action icons get the outline only and the full title not? Or should the whole title get an outline + the actions? Or the whole title + all the actions except for the collapse icon since that already gets outlined as the whole panel title? And if only the action buttons get the outline, should the whole title be a button at all?

WDYT?

Reply:
Shuffled the tags around a bit now reusing the panel title so that only the action buttons get the outline when tabbing, however, the user can still click anywhere on it to expand/collapse, PTAL.